### PR TITLE
Modify persona page to accommodate new component architecture

### PR DESCRIPTION
**Pull request by kn4a-com**
**Motivation**
This change was made due to the Epsilon architecture changes.

### CODE REVIEW

File: Epsilon/Service/CompetenceDocumentService.cs [hunk 1]
```diff
@@ -27,7 +27,7 @@ public async Task<Stream> WriteDocument(Stream stream)
         using var document = WordprocessingDocument.Create(stream, WordprocessingDocumentType.Document);
 
         document.AddMainDocumentPart();
-        document.MainDocumentPart!.Document = _wordDocumentBuilder.Build(components);
+        document.MainDocumentPart!.Document = _wordDocumentBuilder.Build(document.MainDocumentPart, components);
```
Comment (NealGeilen) [inline Epsilon/Service/CompetenceDocumentService.cs]: Why is it needed to change the architecture? Cant you create an open XML element and just return that?

**Review by NealGeilen — CHANGES_REQUESTED** [review]

**Comment by NealGeilen** [comment]
Also please look at what GiHub CodeQL mentions and if that's possible to fix/improve

File: Epsilon/Service/CompetenceDocumentService.cs (hunk 1 above)
Comment (Typiqally) [reply Epsilon/Service/CompetenceDocumentService.cs]: This was required due to the fact that we needed to call `mainDocumentPart.AddAlternativeFormatImportPart(AlternativeFormatImportPartType.Html, altChunkId);`. I'll have another look though

Comment (Typiqally) [reply Epsilon/Service/CompetenceDocumentService.cs]: For now there is no ideal fix, only proper solution I can find is to rename the function to AddToDocument or something similar indicating its usage more properly.

File: Epsilon/Component/WordDocumentBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+{
+    public Document Build(MainDocumentPart mainDocumentPart, IEnumerable<ICompetenceWordComponent> components)
+    {
+        var document = new Document(new Body());
+
+        foreach (var component in components)
+        {
+            var element = component.ToWord(mainDocumentPart);
+            document.Append(element);
+        }
```
[the remark this answers is not in the document]
Comment (Typiqally) [reply Epsilon/Component/WordDocumentBuilder.cs]: Fixed

File: Epsilon.Abstractions/Component/PersonaPage.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+[CompetenceComponentName("persona")]
+public record PersonaPage(string PersonaHtml) : ICompetenceWordComponent
+{
+    public OpenXmlElement ToWord(MainDocumentPart mainDocumentPart)
+    {
+        var body = new Body();
+
+        const string altChunkId = "PersonaPage";
+
+        var personaHtmlStream = new MemoryStream(Encoding.UTF8.GetPreamble().Concat(Encoding.UTF8.GetBytes($"<html>{PersonaHtml}</html>")).ToArray());
```
[the remark this answers is not in the document]
Comment (Typiqally) [reply Epsilon.Abstractions/Component/PersonaPage.cs]: Fixed

**Pull request merged**